File: 고득점 kit/폰켓몬.cpp
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> nums) {
    int answer = 0;
    int N= nums.size() / 2; // 길의 반
    
    // 중복을 없애기 위해 집합 이용
    set<int> selected;
    
    // 싹다 선택
    for (auto n: nums) {
        selected.insert(n);
    }
    
    answer = selected.size() <= N ? selected.size() : N;
    return answer;
}

File: 고득점 kit/해시/폰켓몬/폰켓몬.swift
import Foundation

func solution(_ nums:[Int]) -> Int {
    var answer:Int = 0
    let N:Int = nums.count/2 // 길의 반

    // 집합 이용
    var selected:Set<Int> = []

    for n in nums {
        selected.insert(n)
    }

    answer = selected.count <= N ? selected.count : N
    return answer
}